src/components/Header/Header.tsx

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import * as todoSevice from '../../api/todos';
+import { Errors } from '../../types/Errors';
+import { handleRequestError } from '../../utils/handleRequestError';
+import { useTodosContext } from '../../utils/useTodosContext';
+
+export const Header: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    setTempTodo,
+    setError,
+    setLoadingTodoIds,
+    activeTodos,
+    completedTodos,
+    isLoading,
+    setIsloading,
+  } = useTodosContext();
+  const addTodoInputRef = useRef<HTMLInputElement>(null);
+  const isClassActive = completedTodos.length > 0 && activeTodos.length === 0;
+  const [title, setTitle] = useState('');
+  const [isSubMit, setIsSubmit] = useState(false);
+
+  useEffect(() => {
+    if (isLoading && addTodoInputRef.current) {
+      addTodoInputRef.current.focus();
+      setIsloading(false);
+    }
+  }, [setIsloading, isLoading]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setError(Errors.default);
+    setTitle(event.target.value);
+  };
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
```

Comment:
Choose one style for your functions. Because it's a bad practice to combine different styles